### PR TITLE
Enforce C11/C++11 on Clang, too.

### PR DESCRIPTION
**Pull request by NiLuJe**

### CODE REVIEW

File: Makefile.defs [hunk 1]
```diff
@@ -445,8 +445,14 @@ ifeq "$(CC_IS_CLANG)" "0"
 else
 	# NOTE: Clang's default std value is... iffy. It appears to behave on Linux, but not on Darwin.
 	#       https://clang.llvm.org/cxx_status.html
-	CSTD_FLAGS:=-std=c11
-	CXXSTD_FLAGS:=-std=c++11
+	ifdef DARWIN
+		CSTD_FLAGS:=-std=c11
+		CXXSTD_FLAGS:=-std=c++11
+	else
+		# Keep GNU extensions enabled on Linux (mainly because the CI's Clang version sucks so very hard).
```
Comment (Frenzie) [inline Makefile.defs]: To be clear, that's Clang 8, the latest - 1. I'll probably update it to 9 sometime. Not some curious archeological find from the depths of Ubuntu 16.04. :-P

Comment (NiLuJe) [reply Makefile.defs]: Are you sure it's Clang 8? o_O.

FWIW, luajit builds just fine w/ c11 on Linux w/ Clang 9, and macOS w/ Xcode 11. -_-".

Comment (NiLuJe) [reply Makefile.defs]: Unless Ubuntu mangles LLVM/Clang beyond recognition, which wouldn't surprise me all that much ;).

Comment (Frenzie) [reply Makefile.defs]: Quite sure:

```
$ docker run koreader/kobase-clang:0.1.2 clang --version
clang version 8.0.1-svn363027-1~exp1~20190611212422.76 (branches/release_80)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

Also: #944

Comment (Frenzie) [reply Makefile.defs]: Doesn't matter if Ubuntu does or doesn't; it comes straight from the horse's mouth:

https://github.com/koreader/virdevenv/blob/aea69d62b158dbfbf925e8fa22757848861d5d85/docker/ubuntu/baseimage-clang/clang8.list

Comment (NiLuJe) [reply Makefile.defs]: *sadface*

Comment (Frenzie) [reply Makefile.defs]: https://github.com/koreader/virdevenv/pull/45 is ready to go

File: Makefile.defs [hunk 2]
```diff
@@ -449,7 +449,7 @@ else
 		CSTD_FLAGS:=-std=c11
 		CXXSTD_FLAGS:=-std=c++11
 	else
-		# Keep GNU extensions enabled on Linux (mainly because the CI's Clang version sucks so very hard).
+		# Keep GNU extensions enabled on Linux, otherwise Clang < 9 fails to build LuaJIT.
```
Comment (Frenzie) [inline Makefile.defs]: But how do we know that's an accurate comment? :-P

Comment (NiLuJe) [reply Makefile.defs]: https://circleci.com/gh/NiLuJe/koreader-base/1054?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link ;p

Comment (Frenzie) [reply Makefile.defs]: Rather, I mean who says Clang 9 does?

Comment (NiLuJe) [reply Makefile.defs]: I do ;p.

```
clang  -O2 -fomit-frame-pointer -Wall  -std=c11 -march=native -pipe -O2 -fomit-frame-pointer -frename-registers -fweb -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -U_FORTIFY_SOURCE  -DLUA_MULTILIB=\"lib\" -fno-stack-protector   -c -o ljamalg.o ljamalg.c
clang-9: warning: optimization flag '-frename-registers' is not supported [-Wignored-optimization-argument]
clang-9: warning: optimization flag '-fweb' is not supported [-Wignored-optimization-argument]
gcc-ar rcus 2>/dev/null libluajit.a lj_vm.o ljamalg.o
clang  -O2 -fomit-frame-pointer -Wall  -std=c11 -march=native -pipe -O2 -fomit-frame-pointer -frename-registers -fweb -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -U_FORTIFY_SOURCE  -DLUA_MULTILIB=\"lib\" -fno-stack-protector    -Wl,-O1 -Wl,--sort-common -Wl,--hash-style=gnu -Wl,--as-needed -shared -fPIC -Wl,-soname,libluajit-5.1.so.2   -o libluajit.so lj_vm_dyn.o ljamalg_dyn.o -lm -ldl  
gcc-ranlib libluajit.a
strip --strip-unneeded libluajit.so
clang  -O2 -fomit-frame-pointer -Wall  -std=c11 -march=native -pipe -O2 -fomit-frame-pointer -frename-registers -fweb -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -U_FORTIFY_SOURCE  -DLUA_MULTILIB=\"lib\" -fno-stack-protector    -Wl,-O1 -Wl,--sort-common -Wl,--hash-style=gnu -Wl,--as-needed  -Wl,-E   -o luajit luajit.o libluajit.a -lm -ldl  
strip --strip-unneeded luajit
make[2]: Leaving directory '/var/tmp/niluje/luajit-2.0/src'
```

**Pull request merged**